src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionListBuilder_Sync.cs

```diff
@@ -236,6 +266,24 @@ internal static partial class CompletionListBuilder
 
                 return ($"{beforeText}$0{afterText}", InsertTextFormat.Snippet);
             }
+
+            static void HandleNonIntersectingEdit(SourceText sourceText, ref List<LinePositionSpanTextChange>? additionalTextEdits, ref int? adjustedNewPosition, TextChange textChange)
```

Comment:
I don't think you need `ref` on the list?

Reply:
another thing that jumps out to me, is that it doesn't seem like we are consistent here. In one place we have `getPossiblySnippetizedInsertText` which returns a tuple and doesn't modify references, the locals get reassigned based on what was returned. 
and the new one `HandleNonIntersectingEdit` returns void and operates on references. I think it would be good to align either way. 

also, one is pascal cased the other is camel cased

Reply:
I'll rename the function, but I don't think that any other changes are necessary. The `ref` is necessary because the edits list gets initialized if it's not already initialized. The local function is also different from `getPossiblySnippetizedInsertText`, and I think that's fine. One of them is a static input/output function, the other is a handler that has side effects. I don't think it's worth it try and make this function an input/output: that would make a good chunk of the code sharing between the two locations not possible.

Reply:
my physical brain prefers to think without side effects `f : x -> y` but OK 😄